src/cockpit/channels/dbus.py

```diff
@@ -172,11 +171,54 @@ class DBusChannel(Channel):
     matches = None
     name = None
     bus = None
+    owner = None
 
     # This needs to be a fair mutex so that outgoing messages don't
     # get re-ordered.  asyncio.Lock is fair.
     watch_processing_lock = asyncio.Lock()
 
+    def add_match(self, rule, handler):
+        def sync_handler(message):
+            task = asyncio.create_task(handler(message))
+            self.tasks.add(task)
+            task.add_done_callback(self.tasks.discard)
+        self.matches.append(self.bus.add_match(rule, sync_handler))
+
+    async def setup_name_owner_tracking(self):
+        def send_owner(owner):
+            # We must be careful not to send duplicate owner
+            # notifications. cockpit.js relies on that.
+            if self.owner != owner:
+                self.owner = owner
+                self.send_message(owner=owner)
+        async def handler(message):
```

Comment:
I don't understand why this needs to be async.  It doesn't await anything.

Reply:
It's async because self.add_match below expects a async function. I did that because it is slightly more convenient for other users of self.add_match, but it's not clean, agreed.

Reply:
I have changed this to use self.bus.add_match directly.